### PR TITLE
🏃[e2e] Reorg helpers func

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR is a big reshuffle of all the helpers defined in the E2E test framework so they are now consistently organized "by objects" (more detail in the issue)

No other changes were applied, except adding `// Deprecated` to few helpers that are not anymore in line with the recent changes in the test framework (eg.  `AssertAllClusterAPIResourcesAreGone `  `ApplyYAMLURL` are now deprecated).

All the deprecated helpers are now in a single `deprecated.go` file.

**Which issue(s) this PR fixes**:
Fixes https://github.com/kubernetes-sigs/cluster-api/issues/2921

xref #2753
xref #2637
xref #2636

/assign @vincepri
/assign @sedefsavas

### CODE REVIEW

File: test/framework/deprecated.go [hunk 1]
```diff
@@ -17,22 +17,37 @@ limitations under the License.
...
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
-
 	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/cluster-api/test/framework/management/kind"
+	"sigs.k8s.io/cluster-api/test/framework/options"
+
+	corev1 "k8s.io/api/core/v1"
```
Comment (benmoss) [inline test/framework/deprecated.go:39]: corev1 and bootstrapv1 are duplicate imports

Comment (fabriziopandini) [reply test/framework/deprecated.go:39]: fixed!

**Comment by fabriziopandini** [comment]
@benmoss thanks for the feedback! fixed!

**Comment by vincepri** [comment]
@fabriziopandini 
```
sigs.k8s.io/cluster-api/test/framework
  Incompatible changes:
  - WatchControllerLogs: removed
  - WatchControllerLogsInput: removed
```

**Comment by fabriziopandini** [comment]
@vince Those functions were added by https://github.com/kubernetes-sigs/cluster-api/pull/2911 yesterday;
I'm renaming them into `WatchDeploymentLogs` and `WatchDeploymentLogsInput` because this is what they are really doing (nothing controller specific)

**Comment by vince** [comment]
@fabriziopandini wrong Vince!

**Pull request merged**